### PR TITLE
Implements #45 - Cipher ZK encrypt and decrypt

### PR DESCRIPTION
**Pull request by vlopes11**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ kelvin = "0.18"
 nstack = "0.4"
 lazy_static = "1.3.0"
 hades252 = { git = "https://github.com/dusk-network/hades252", tag = "v0.7.0" }
-dusk-plonk = "0.2.7"
+dusk-plonk = "0.2.8"
```
Comment (CPerezz) [inline Cargo.toml:14]: Couldn't we set `dusk-plonk = "0.2"` so it'll always take the latest one avobe?

**Review by CPerezz — APPROVED** [review]
LGTM. 

Just the nit/question of the version. 
Also, please, bump up the "patch", update the `CHANGELOG` and ship a new GH release.

Feel free to merge once both things are addressed.

File: Cargo.toml (hunk 1 above)
Comment (vlopes11) [reply Cargo.toml:14]: I think its not desirable considering plonk is `MAJOR=0`, so the patch is in practice the minor version.

If we go to version `1` of plonk, then that will be fine. Before that, any minor version upgrade will be enforced and these sometimes can be breaking

Comment (CPerezz) [reply Cargo.toml:14]: Makes sense. Bump up the version of Poseidon anyway.

**Pull request merged**